src/ProtocolHandler/HTTP/Responses/Response.cs

```diff
@@ -0,0 +1,79 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Chorizo.Date;
+using Chorizo.Sockets.CzoSocket;
+
+namespace Chorizo.ProtocolHandler.HTTP.Responses
+{
+    public class Response : IResponse
+    {
+        private readonly IChorizoSocket _socket;
+        private readonly IDateTimeProvider _dateTimeProvider;
+        private readonly ResponseCodes _responseCodes;
+        private string _protocol = "HTTP/1.1";
+        private int _statusCode = 200;
+        private string _statusText = "OK";
+        private readonly Dictionary<string, string> _headers;
+        private readonly Dictionary<string, string> _additionalHeaders;
+
+        public Response(IChorizoSocket chorizoSocket, IDateTimeProvider dateTimeProvider = null)
+        {
+            _socket = chorizoSocket;
+            _dateTimeProvider = dateTimeProvider ?? new DateTimeProvider();
+            _headers = new Dictionary<string, string>();
+            _additionalHeaders = new Dictionary<string, string>();
+            _responseCodes = new ResponseCodes();
+        }
+
+        public IResponse Append(string field, string value)
+        {
+            _additionalHeaders.Add(field, value);
+            return this;
+        }
+
+        public IResponse Status(int responseCode)
+        {
+            _statusCode = responseCode;
+            _statusText = _responseCodes.getMessage(responseCode);
+            return this;
+        }
+
+        public void Send(string message = null, string contentType = "text/html")
```

Comment:
I found this function hard to read. Can you break it down into small functions? Additionally, It doesn't make complete sense to me as to why `Send` should live in the `Response`. I see `Response` as an encoding of an `HTTP Response` that doesn't have knowledge of how its sent back to the client (this should be handled elsewhere). This may make more sense as you start building out more features of your server.